metadata/job.go

```diff
@@ -131,6 +132,31 @@ func (m Metadata) Strings() map[string]string {
 	return bm
 }
 
+// ConvertToListOfStrings convert an interface to list of strings, support if the type of the interface
+// is list of strings or list of interfaces, where each interface in the list is a string.
+func ConvertToListOfStrings(rawInterface interface{}) ([]string, error) {
```

Comment:
TODO: add UT

Reply:
Please make this a member of the Metadata struct, maybe MultiString instead of String?

Reply:
Made as Metadata struct function, also kept the logic on the other functions there, which return value and found for key.
It's might cause issues, because the user will not be able to distinguish between an empty list, or wasn't able to convert the list of interfaces into list of strings. 
WDYT? 

I added UT's.
Not sure about the MultiString you mean for the DocString? you mean maybe Multilines?